tests/run/cppclass_constructor_segfault.pyx

```diff
@@ -0,0 +1,43 @@
+pyo = "test"
+
+cdef:
+	cppclass test:
+		test():
+			global pyo # Does not cause segfault
+
+			# Causes segfault
+			#[1, 2, 3]
+			#i = 1000000000000000000000
+			#int(123)
+			#print(123)
+			#hi = "test"
+			#pyo
+			#str("test")
+			#import os
+			#hex(40)
+			#chr(40)
+			#bin(40)
+			#range(40)
+			#5 / 0
+
+			# No segfault, no error
+			#"test"
+			#str()
+			#2 ** 900
+			#None
+			#False
+			#u"Test"
+			#ord(b't')
+			#abs(40)
+			#type(40)
+			#bool(40)
+
+			# Fatal Python error: PyEval_SaveThread: NULL tstate
+			#with nogil:
+			#	pass
+
+			# Fatal Python error: PyThreadState_Get: no current thread
+			#bytearray()
+			#bytes()
+
+	test t = test()
```

Comment:
The problem here is that the line is split into

```
test t; // run the instant the module loads, in a state possibly without the GIL, almost certainly before pyo is set to up

t = test(); // run later, with the GIL during the module setup
```

I.e. the 0-argument constructor is called twice. This basically happens any time you use a c++ object and needs a significant change in how Cython stores C++ objects to fix.

https://github.com/cython/cython/pull/4225 should eventually fix it, but isn't even close to don't at the moment. For now you need to heap allocate.